sphinx_needs/lsp/esbonio.py

```diff
@@ -111,9 +142,13 @@ def hover(self, context: HoverContext) -> str:
     def definition(self, context: DefinitionContext) -> List[Location]:
         """Return location of definition of a need."""
         if isinstance(self.rst, SphinxLanguageServer) and self.rst.app:
+            # get and check needs.json path
+            needs_json = get_needs_json(self.rst.app)
+            if not needs_json or not needs_json.exists():
+                self.logger.warning(f"needs.json {needs_json} not exists. No Sphinx-Needs language features activated.")
```

Comment:
deja vu. This is the same code as above, no? is it necessary to repeat the logic, or can this be reused (DRY)?

Reply:
unfortunately not. I know the log message it's kind of duplicate. But the line below, return type, is different for different language features in different functions.